nimbleclient/v1/restclient.py

```diff
@@ -203,30 +228,40 @@ def get(self, endpoint, **params):
             paginated_data.extend(response.json()["data"])
             total_rows = response.json()['totalRows']
             retrieved_rows = len(paginated_data)
-            #current_end_row = response.json()['endRow']
+            # current_end_row = response.json()['endRow']
 
             if 'startRow' in params and 'endRow' in params:
                 requested_rows = params['endRow'] - params['startRow']
             elif 'startRow' not in params and 'endRow' not in params:
-                requested_rows = total_rows # Get all available records
+                requested_rows = total_rows  # Get all available records
             else:
                 if 'startRow' in params:
                     requested_rows = total_rows - params['startRow']
                 else:
                     requested_rows = params['endRow']
-            #print("Total records requested:", requested_rows, ", Retrieved records:", retrieved_rows)
+
+            logging.debug("Total records requested: %d, Retrieved records: %d", requested_rows, retrieved_rows)
 
             pending_rows = requested_rows-retrieved_rows
             params.clear()
             while pending_rows > 0:
                 params['startRow'] = response.json()['endRow']
-                #print("Retrieved:", retrieved_rows, "Pending:", pending_rows, "Next startRow:", params['startRow'])
-                response = requests.get(
-                    url,
-                    params=params,
-                    headers=self._headers,
-                    verify=False,
-                )
+                logging.debug("Retrieved: %d, Pending: %d, Next startRow: %d", retrieved_rows, pending_rows, params['startRow'])
+                if filter is not None:
+                    response = requests.post(
+                        url,
+                        params=params,
+                        json=self.build_advanced_criteria(filter),
+                        headers=self._headers,
+                        verify=False
+                    )
+                else:
+                    response = requests.get(
+                        url,
+                        params=params,
+                        headers=self._headers,
+                        verify=False,
+                    )
                 records = response.json()["data"][:pending_rows]
```

Comment:
Doesn't the response have enough information for you to know if there are more results? I would do some testing to verify. It seems to me that the logic for retrieving the paginated data is overly complicated, you should trust that the API gives you what was asked for, if it doesn't that's a bug on the REST API, not our code.

Reply:
Agreed, refactored the get() to simplify the logic.